go.go

```diff
@@ -446,6 +496,27 @@ const goMarshalFieldLen = `{{if eq .Type "bool"}}
 		}
 	}
 {{else if eq .Type "int32"}}
+{{- if .TypeList}}
+	if x := len(o.{{.NameTitle}}); x != 0 {
+		if x > ColferListMax {
+			return 0, ColferMax(fmt.Sprintf("colfer: field {{.String}} exceeds %d elements", ColferListMax))
+		}
+		for l += 2; x >= 0x80; l++ {
+			x >>= 7
+		}
+		for _, v := range o.{{.NameTitle}} {
+			x1 := uint32((v << 1) ^ (v >> 31))
+			l++
```

Comment:
You can do this increment with one instruction outside of the loop as `l += len(o.{{.NameTitle}})`. Should be faster for large slices.

Reply:
Dito for the 64-bit version below.

Reply:
+